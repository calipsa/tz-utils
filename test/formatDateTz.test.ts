import {
  formatDateTz,
  names as timezoneNames,
  deprecatedNames,
  notWorkingInDateFnsNames,
} from '../src'

import dates from './data/dates'

const FORMAT = 'yyyy-MM-dd, HH:mm:ss'

const timezones = [
  ...timezoneNames,
  ...deprecatedNames,
  ...notWorkingInDateFnsNames,
]

describe('formatDateTz', () => {
  for (const tz of timezones) {
    for (const date of dates) {
      test(`splitDate (${date}, ${tz})`, () => {
        const result = formatDateTz(date, FORMAT, tz)
        expect(result).toMatchSnapshot()
      })
    }
  }

  for (const date of dates) {
    test(`splitDate deprecated tz (${date})`, () => {
      const res1 = formatDateTz(date, FORMAT, 'Navajo')
      const res2 = formatDateTz(date, FORMAT, 'America/Denver')
      expect(res1).toBe(res2)
    })
  }

  for (const date of dates) {
    test(`splitDate not working in date-fns tz (${date})`, () => {
      const res1 = formatDateTz(date, FORMAT, 'Etc/GMT-3')
      const res2 = formatDateTz(date, FORMAT, 'Europe/Moscow')
      expect(res1).toBe(res2)
    })
  }
})
