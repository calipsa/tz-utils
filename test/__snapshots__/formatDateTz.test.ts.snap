// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatDateTz splitDate (2019-11-15T23:45:45.392Z, Europe/London) 1`] = `"2019-11-15, 23:45:45"`;

exports[`formatDateTz splitDate (2019-11-15T23:45:45.392Z, Europe/Moscow) 1`] = `"2019-11-16, 02:45:45"`;

exports[`formatDateTz splitDate (2019-11-15T23:45:45.392Z, Europe/Paris) 1`] = `"2019-11-16, 00:45:45"`;

exports[`formatDateTz splitDate (2019-11-15T23:45:45.392Z, UTC) 1`] = `"2019-11-15, 23:45:45"`;

exports[`formatDateTz splitDate (2019-11-16T00:43:45.392Z, Europe/London) 1`] = `"2019-11-16, 00:43:45"`;

exports[`formatDateTz splitDate (2019-11-16T00:43:45.392Z, Europe/Moscow) 1`] = `"2019-11-16, 03:43:45"`;

exports[`formatDateTz splitDate (2019-11-16T00:43:45.392Z, Europe/Paris) 1`] = `"2019-11-16, 01:43:45"`;

exports[`formatDateTz splitDate (2019-11-16T00:43:45.392Z, UTC) 1`] = `"2019-11-16, 00:43:45"`;

exports[`formatDateTz splitDate (2019-11-16T00:44:45.392Z, Europe/London) 1`] = `"2019-11-16, 00:44:45"`;

exports[`formatDateTz splitDate (2019-11-16T00:44:45.392Z, Europe/Moscow) 1`] = `"2019-11-16, 03:44:45"`;

exports[`formatDateTz splitDate (2019-11-16T00:44:45.392Z, Europe/Paris) 1`] = `"2019-11-16, 01:44:45"`;

exports[`formatDateTz splitDate (2019-11-16T00:44:45.392Z, UTC) 1`] = `"2019-11-16, 00:44:45"`;

exports[`formatDateTz splitDate (2019-11-16T00:45:45.392Z, Europe/London) 1`] = `"2019-11-16, 00:45:45"`;

exports[`formatDateTz splitDate (2019-11-16T00:45:45.392Z, Europe/Moscow) 1`] = `"2019-11-16, 03:45:45"`;

exports[`formatDateTz splitDate (2019-11-16T00:45:45.392Z, Europe/Paris) 1`] = `"2019-11-16, 01:45:45"`;

exports[`formatDateTz splitDate (2019-11-16T00:45:45.392Z, UTC) 1`] = `"2019-11-16, 00:45:45"`;

exports[`formatDateTz splitDate (2019-11-16T23:45:45.392Z, Europe/London) 1`] = `"2019-11-16, 23:45:45"`;

exports[`formatDateTz splitDate (2019-11-16T23:45:45.392Z, Europe/Moscow) 1`] = `"2019-11-17, 02:45:45"`;

exports[`formatDateTz splitDate (2019-11-16T23:45:45.392Z, Europe/Paris) 1`] = `"2019-11-17, 00:45:45"`;

exports[`formatDateTz splitDate (2019-11-16T23:45:45.392Z, UTC) 1`] = `"2019-11-16, 23:45:45"`;
